packages/next/pages/_document.tsx

```diff
@@ -186,7 +186,15 @@ export default class Document<P = {}> extends Component<DocumentProps & P> {
 
 // Add a speical property to the built-in `Document` component so later we can
 // identify if a user customized `Document` is used or not.
-;(Document as any).__next_internal_document = true
+;(Document as any).__next_internal_document = () => (
```

Comment:
Can we define it as a named function then assign it to Document later? would be helpful for development

```js
function FunctionalDocument() {...}

;(Document as any).FunctionalDocument = FunctionalDocument
```